dotnet/samples/KernelSyntaxExamples/Example25_ReadOnlyMemoryStore.cs

```diff
@@ -148,11 +147,9 @@ public IAsyncEnumerable<string> GetCollectionsAsync(CancellationToken cancellati
                 }
             }
 
-            embeddings.SortByScore();
-
-            foreach (var item in embeddings)
+            foreach (var item in embeddings.OrderByDescending(l => l.Item2).Take(limit))
             {
-                yield return (item.Value, item.Score.Value);
+                yield return (item.Item1, item.Item2);
```

Comment:
Use the tuple for clarity where possible. Not sure if you are able to use it in the OrderDescendingBy.

```suggestion
            foreach (var (memoryRecord, similarity) in embeddings.OrderByDescending((mr, similarity) => similarity).Take(limit))
            {
                yield return (memoryRecord, similarity);
```